src/auth/integration-tests/src/lib.rs

```diff
@@ -115,3 +120,118 @@ pub async fn api_key() -> anyhow::Result<()> {
 
     Ok(())
 }
+
+pub async fn workload_identity_provider_url_sourced() -> anyhow::Result<()> {
+    let project = std::env::var("GOOGLE_CLOUD_PROJECT").expect("GOOGLE_CLOUD_PROJECT not set");
+    let audience = get_oidc_audience();
+    let service_account = get_byoid_service_account();
+    let service_account_data =
+        serde_json::from_value::<HashMap<String, String>>(service_account.clone())
+            .expect("failed to read service account data as map");
+    let client_email = service_account_data
+        .get("client_email")
+        .expect("missing client_email");
+
+    let id_token =
+        generate_id_token(audience.clone(), client_email.to_string(), service_account).await?;
+
+    let source_token_response_body = serde_json::json!({
+        "id_token": id_token,
+    })
+    .to_string();
+
+    let server = Server::run();
```

Comment:
BTW, you still owe me a doc explaining why this is the better choice for faking HTTP servers.

Reply:
yes, gonna start it this week.